### PR TITLE
Fixed #34542 -- Made createsuperuser handle required blank fields in non-interactive mode.

### PR DESCRIPTION
**Pull request by xYazz**
Ticket: [#34542](https://code.djangoproject.com/ticket/34542)

### CODE REVIEW

File: tests/auth_tests/test_management.py [hunk 1]
```diff
@@ -968,6 +968,34 @@ def test_blank_username_non_interactive(self):
...
+        call_command(
+            "createsuperuser",
+            email="",
+            username="joe",
+            interactive=False,
+            stdin=MockTTY(),
+            stdout=new_io,
+            stderr=new_io,
+        )
+        self.assertEqual(new_io.getvalue().strip(), "Superuser created successfully.")
```
Comment (nessita) [inline tests/auth_tests/test_management.py:983]: I think these two tests would need a few extra assertions ensuring the superuser was actually created (besides the existing assert over the returned success message. See `test_basic_usage` as example of the assertions we could use.

**Review by nessita — COMMENTED** [review]
Looks great! I added a small suggestion. Thanks!

File: django/contrib/auth/management/commands/createsuperuser.py [hunk 2]
```diff
@@ -219,11 +219,16 @@ def handle(self, *args, **options):
                 for field_name in self.UserModel.REQUIRED_FIELDS:
                     env_var = "DJANGO_SUPERUSER_" + field_name.upper()
                     value = options[field_name] or os.environ.get(env_var)
+                    field = self.UserModel._meta.get_field(field_name)
                     if not value:
+                        if field.blank and (
+                            options[field_name] is not None
+                            or os.environ.get(env_var) is not None
+                        ):
+                            continue
```
Comment (felixxm) [inline django/contrib/auth/management/commands/createsuperuser.py:228]: I find this construction unusual, can we get rid of nested `if`?

Comment (xYazz) [reply django/contrib/auth/management/commands/createsuperuser.py:228]: In short, I'm not so sure.
It's necessary to check if `value` is None and when it is, it requires additional checks for `field.blank` and if either of `options[field_name]`, `os.environ.get(env_var)` aren't None to determine if the command was used correctly.

I've tried to get rid of the nested `if` but I'm not convinced that the changed code is in any way better.

Or if you've meant moving `(options[field_name] is not None or os.environ.get(env_var) is not None`) to a variable and then using that variable in the `if` then sure.

Comment (felixxm) [reply django/contrib/auth/management/commands/createsuperuser.py:228]: The only possible value is an empty string. IMO, it'd be clearer to use it.
```suggestion
                        if field.blank and (
                            options[field_name] == "" or os.environ.get(env_var) == ""
                        ):
                            continue
```

**Review by felixxm — COMMENTED** [review]
@xYazz Thanks :+1: Welcome aboard :boat:

**Pull request merged**